validation/src/test/scala/hmda/validation/rules/lar/validity/V290Spec.scala

```diff
@@ -0,0 +1,40 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.validation.dsl.{ Failure, Success }
+import hmda.validation.rules.lar.LarEditCheckSpec
+
+class V290Spec extends LarEditCheckSpec {
+
+  property("Succeeds when valid MSA/MD, state and county codes are present") {
+    forAll(larGen) { lar =>
+      val validGeography = lar.geography.copy(msa = "17020", county = "007", state = "06")
+      val validLar = lar.copy(geography = validGeography)
+      V290(validLar) mustBe a[Success]
+    }
+  }
+
+  property("Succeeds when MSA == NA") {
+    forAll(larGen) { lar =>
+      val validGeography = lar.geography.copy(msa = "NA")
+      val validLar = lar.copy(geography = validGeography)
+      V290(validLar) mustBe a[Success]
+    }
+  }
+
+  property("Succeeds when code is MD (not MSA) with valid state and county combination") {
+    forAll(larGen) { lar =>
+      val validGeography = lar.geography.copy(msa = "48424", state = "12", county = "086")
+      val validLar = lar.copy(geography = validGeography)
+      V290(validLar) mustBe a[Success]
```

Comment:
After checking with @lesgou, the logic should be:

1. If MSA/MD = "NA", edit check passes
2. If MSA/MD != "NA, check for valid (state, county, msa) combination or valid (state, county, md) combination
I've updated the logic to reflect this